clang/lib/Sema/SemaChecking.cpp

```diff
@@ -3233,6 +3233,17 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
     if (BuiltinCountZeroBitsGeneric(*this, TheCall))
       return ExprError();
     break;
+
+  case Builtin::BI__builtin_allow_runtime_check: {
+    Expr *Arg = TheCall->getArg(0);
+    // Check if the argument is a string literal.
+    if (!isa<StringLiteral>(Arg->IgnoreParenImpCasts())) {
```

Comment:
This might be a reasonable restriction, but consider other cases like: `constexpr` variable of string type, `constexpr` function call that returns a `const char *`, `_Generic` that results in a string constant, etc. Would those be reasonable to support?

Reply:
My interest is in using this feature to implement `assert` like macros which will be enabled in cold code, and optimized out in hot.  E.g. MY_ASSERT1, MY_ASSERT2... 
Each macro can provide own kind as literal. Number of different kinds is supposed to be limited. So literal looks enough for me.
We can relax in the future, restricting could be harder.

Reply:
Resolving in favor of the longer thread above.